From 8bbb79d3b9b846428304cdb05db19628f61dc8a3 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Fri, 22 Mar 2013 20:49:25 -0700
Subject: [PATCH 07/10] ck: discourage 250HZ as default

Imported from: http://ck.kolivas.org/patches/3.0/3.8/3.8-ck1/patches/hz-no_default_250.patch

Make 250HZ not be the default to discourage desktop users from choosing this
option since 1000 will provide better latencies with only miniscule amounts
of extra overhead and power consumption.

-ck

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 kernel/Kconfig.hz |   17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/kernel/Kconfig.hz b/kernel/Kconfig.hz
index b0c107e..f67cd4a 100644
--- a/kernel/Kconfig.hz
+++ b/kernel/Kconfig.hz
@@ -23,13 +23,14 @@ choice
 	  with lots of processors that may show reduced performance if
 	  too many timer interrupts are occurring.
 
-	config HZ_250
+	config HZ_250_NODEFAULT
 		bool "250 HZ"
 	help
-	 250 Hz is a good compromise choice allowing server performance
-	 while also showing good interactive responsiveness even
-	 on SMP and NUMA systems. If you are going to be using NTSC video
-	 or multimedia, selected 300Hz instead.
+	 250 HZ is a lousy compromise choice allowing server interactivity
+	 while also showing desktop throughput and no extra power saving on
+	 laptops. No good for anything.
+
+	 Recommend 100 or 1000 instead.
 
 	config HZ_300
 		bool "300 HZ"
@@ -43,14 +44,16 @@ choice
 		bool "1000 HZ"
 	help
 	 1000 Hz is the preferred choice for desktop systems and other
-	 systems requiring fast interactive responses to events.
+	 systems requiring fast interactive responses to events. Laptops
+	 can also benefit from this choice without sacrificing battery life
+	 if dynticks is also enabled.
 
 endchoice
 
 config HZ
 	int
 	default 100 if HZ_100
-	default 250 if HZ_250
+	default 250 if HZ_250_NODEFAULT
 	default 300 if HZ_300
 	default 1000 if HZ_1000
 
-- 
1.7.10.4

